libsrc/meshing/basegeom.cpp

```diff
@@ -1165,4 +1162,10 @@ namespace netgen
   }
 
   static RegisterClassForArchive<NetgenGeometry> regnggeo;
+
+  GeometryRegisterArray& FetchGeometryRegisterArray (){
+    static GeometryRegisterArray *geometryregister = new GeometryRegisterArray();
+    return *geometryregister;
```

Comment:
No need to heap-allocate here, a static variable is fine.
```suggestion
    static GeometryRegisterArray geometryregisterarrayInstance;
    return &geometryregisterarrayInstance;
```